dags/veda_data_pipeline/groups/discover_group.py

```diff
@@ -42,60 +43,59 @@ def get_files_to_process(ti):
     """Get files from S3 produced by the discovery task.
     Used as part of both the parallel_run_process_rasters and parallel_run_process_vectors tasks.
     """
-    payload = ti.xcom_pull(task_ids=f"{group_kwgs['group_id']}.discover_from_s3")
-    payloads_xcom = payload.pop("payload", [])
+    dynamic_group_id = ti.task_id.split(".")[0]
+    payload = ti.xcom_pull(task_ids=f"{dynamic_group_id}.discover_from_s3")
+    payloads_xcom = payload[0].get("payload", [])
     dag_run_id = ti.dag_run.run_id
     for indx, payload_xcom in enumerate(payloads_xcom):
         time.sleep(2)
         yield {
             "run_id": f"{dag_run_id}_{uuid.uuid4()}_{indx}",
-            **payload,
+            **payload[0],
             "payload": payload_xcom,
         }
```

Comment:
By default, this xcom will load lazily as a sequence, and needs to be expanded before indexing.

Reply:
:I_have_a_dumb_question: does xcom mean cross-communication?

Reply:
Yep :) https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/xcoms.html